pandas/core/nanops.py

```diff
@@ -718,6 +718,8 @@ def reduction(values, axis=None, skipna=True, mask=None):
                 result = np.nan
         else:
             result = getattr(values, meth)(axis)
+            if is_integer(result) and result == _int64_max:
```

Comment:
It looks affect to integer dtype, `pd.Series([_int64_max]).min() / max()`?

Reply:
Fixed, now it only applies the conversion from `_int64_max` to `NaT` if given an appropriate dtype.